collector/util.go

```diff
@@ -16,10 +16,26 @@ package collector
 
 import "github.com/google/cadvisor/container"
 
-func (endpointConfig *EndpointConfig) configure(containerHandler container.ContainerHandler) {
+var DefaultURLConfig = URLConfig{
+	Protocol: "http",
+	Port:     "8000",
+}
+
+func (endpointConfig *EndpointConfig) configure(containerHandler container.ContainerHandler, defaultURLConfig URLConfig) {
 	//If the exact URL was not specified, generate it based on the ip address of the container.
 	endpoint := endpointConfig
 	if endpoint.URL == "" {
+
+		if endpoint.URLConfig.Protocol == "" {
```

Comment:
-1 for these defaults. I'd prefer the per-collector factories fail instead of applying global defaults.